snap/snapcraft.yaml

```diff
@@ -1,81 +1,61 @@
-name: saber # you probably want to 'snapcraft register <name>'
-base: core22 # the base snap is the execution environment for this snap
+name: saber
+base: core24
 adopt-info: saber
-grade: stable # must be 'stable' to release into candidate/stable channels
-confinement: strict # use 'strict' once you have the right plugs and slots
+grade: stable
+confinement: strict
 license: GPL-3.0
 compression: lzo
-icon: assets/icon/icon.png
-donation: https://github.com/sponsors/adil192
-source-code: https://github.com/saber-notes/saber
-issues: https://github.com/saber-notes/saber/issues
-contact: soumyadeepghosh2004@zohomail.in
-architectures:
-  - build-on: amd64
-  - build-on: arm64
-  
+icon: assets/icon/icon.svg
+platforms:
+  amd64:
+  arm64:
+
 parts:
   rustup:
-    plugin: nil
-    build-packages: [wget]
-    build-environment:
-      - RUSTUP_HOME: $CRAFT_PART_INSTALL/usr/share/rust
-      - CARGO_HOME: $CRAFT_PART_INSTALL/usr/share/rust
-      - CARGO_BUILD_JOBS: $CRAFT_PARALLEL_BUILD_COUNT
-    override-pull: |
-      wget https://sh.rustup.rs -O $CRAFT_PART_SRC/rustup-init.sh
-      chmod +x $CRAFT_PART_SRC/rustup-init.sh
-    override-build: |
-      $CRAFT_PART_SRC/rustup-init.sh -y --no-modify-path
-      mkdir -p $CRAFT_PART_INSTALL/usr/bin
-      for i in `ls $RUSTUP_HOME/bin/`; do
-        ln -s ../share/rust/bin/$i $CRAFT_PART_INSTALL/usr/bin/$i
-      done
-    override-prime: |
-      echo 'Skip Prime'
+    plugin: rust
+    source: .
+    rust-channel: "1.90"
+    override-build: ""
+    override-prime: ""
+
```

Comment:
Will you need to manually update this rust version?

Reply:
The linux github actions workflow doesn't install rust at all. Does the snap build require it?

Reply:
I am not sure about the dependencies. I just updated the existing deps to get them in a better possible way. I didn't cross check the deps.